src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -79,6 +98,29 @@ def extract_strings(self) -> Mapping[int, Set[str]]:
                 )
                 continue
 
+    @metrics.wraps("process_messages.limit_messages")
+    def limit_messages(self, keys_to_remove: Sequence[PartitionIdxOffset]) -> None:
```

Comment:
continuing the batch refactor discussion at https://github.com/getsentry/sentry/pull/38302#discussion_r959891729 here -- the batch now no longer concerns itself with rate limiting, but see the other PR for details